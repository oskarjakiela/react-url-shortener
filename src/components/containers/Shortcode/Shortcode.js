import React from 'react';

import './Shortcode.css';
import { Link, Text } from 'components/ui';


const displayName = 'Shorty-Shortcode';
const domain = 'shooooort.com/';

const Shortcode = ({
  url,
  value,
}) => (
  <div className={displayName}>
    <span className={`${displayName}__url-body`}>
      <span className={`${displayName}__domain`}>
        {domain}
      </span>

      <span className={`${displayName}__value`}>
        {value}
      </span>
    </span>

    <Link className={`${displayName}__copy-link`}>
      Click to copy this link
    </Link>

    <br />

    <Text className={`${displayName}__url`} unimportant>
      {url}
    </Text>
  </div>
);

Shortcode.displayName = displayName;

export default Shortcode;
