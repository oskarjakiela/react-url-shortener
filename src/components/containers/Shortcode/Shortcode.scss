.Shorty-Shortcode {
  line-height: 1.6875rem;

  &:hover {
    .Shorty-Shortcode__copy-link {
      display: inline-block;
    }
  }

  .Shorty-Shortcode__copy-link {
    display: none;
  }
  
  &__url {
    display: inline-block;
    width: 100%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__url-body {
    margin-right: 1rem;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    font-size: 1rem;
    color: #555;
  }

  &__value {
    color: #eb4a42;
  }
}


