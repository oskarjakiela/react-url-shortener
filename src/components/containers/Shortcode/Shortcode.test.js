import React from 'react';
import renderer from 'react-test-renderer';

import Shortcode from './Shortcode';


it('<Shortcode /> renders correctly', () => {
  const tree = renderer
    .create(<Shortcode
      url="https://example.com"
      value="f00ba4"
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
