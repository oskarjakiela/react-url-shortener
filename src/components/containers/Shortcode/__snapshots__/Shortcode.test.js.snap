// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shortcode /> renders correctly 1`] = `
<div
  className="Shorty-Shortcode"
>
  <span
    className="Shorty-Shortcode__url-body"
  >
    <span
      className="Shorty-Shortcode__domain"
    >
      shooooort.com/
    </span>
    <span
      className="Shorty-Shortcode__value"
    >
      f00ba4
    </span>
  </span>
  <a
    className="Shorty-Link Shorty-Shortcode__copy-link"
  >
    Click to copy this link
  </a>
  <br />
  <span
    className="Shorty-Text Shorty-Text--unimportant Shorty-Shortcode__url"
  >
    https://example.com
  </span>
</div>
`;
